keps/sig-cluster-lifecycle/kubeadm/20191229-kubeadm-move-out-of-kk.md

```diff
@@ -0,0 +1,594 @@
+announced as stable. README files in the kubeadm repository
+should denote that.
+
+It is out of scope for this proposal to try to solve the problem.
+
+#### The post-submit job for tagging/branching k/kubeadm fails
+
+In cases the post-submit job fails, maintainers should be able
+to apply the required changes manually. Sufficient number of
+maintainers should have access to perform these actions.
+
+Close to a release, coordination with SIG Release will be required,
+to make sure that the exact, desired SHA of kubeadm is released.
+
+#### A unwanted commit gets tagged in the kubeadm repository
+
+If the post-submit job is soon to tag a commit in k/kubeadm
+for release (e.g. `v1.18.0`), there is a chance that a PR in k/kubeadm
+can merge before this happens and a commit that is not desired
+receives the tag.
+
+This is a mistake that the kubeadm maintainers should not allow.
+Close to a release, there is code freeze at effect, so PRs should
+pass the proper review and possibly be postponed for the next release.
+
+While kubeadm is already quite low volume in commits, coordination with
+SIG Release might be required to know when the release button is pushed
+and to prevent such problems.
+
+#### The kubeadm go.mod includes outdated components
+
+Once kubeadm is decoupled from k/k it will no longer benefit
+from all the updates and security fixes that its components under
+"staging" see. The imports in the kubeadm go.mod will need
+to be updated on a regular basis to mitigate CVEs and critical bugs.
+
+Tools such as https://docs.renovatebot.com/golang/ were
+proposed. Automatic bulk updates can increase the noise in the
+kubeadm repository but always make sure kubeadm depends on the
+latest dependencies. Manual updates can get sloppy and miss critical
+commits.
+
+The current proposal is to rely on manual updates and make sure
+kubeadm is always up to date before a final release, but also invest
+time into automating this process.
+
+Coordination with the security team and relevant SIGs might be
+required.
+
+#### The kubeadm module does not use SEMVER
+
+Users importing the kubeadm module will find that the k8s versioning
+is actually not SEMVER, because breaking changes happen on MINOR
+changes instead of MAJOR ones.
+
+A proposed solution here is to dual-tag commits with a version that
+is SEMVER friendly - e.g.:
+- `v1.17.0` -> `v0.17.0`
+
+Dual-tagging with `v17.0.0` is not an option, because Go modules
+will require the `v17` package.
+
+But the problem remains with `v0.17.0`, because once k8s becomes
+`v2.0.0`, the dual-tags (SEMVER) will overlap with the non-SEMVER tags.
+
+There are no good solutions for this problem and users might have to
+use SHAs to import the kubeadm module.
+
+See the related discussions at:
+https://github.com/kubernetes/kubernetes/issues/72638
+https://github.com/kubernetes/kubernetes/issues/84372
+
+## Design Details
+
+### Test Plan
+
+No changes will be required for testing kubeadm.
```

Comment:
I think that all the v1.18 pre-releases/branch cu should be considered as test of the v1.18 release cut

Reply:
i guess i should mention here that the periodics will not change, but we need to have presubmits for all branches.

Reply:
re-organized the doc to have the "Docs" and "Testing" sections under "Design Details".